src/main/java/org/folio/service/mapping/processor/RuleFactory.java

```diff
@@ -1,56 +1,120 @@
+
+  private static final String SET_VALUE_TRANSLATION = "set_value";
+  private static final String VALUE_PARAMETER = "value";
+  private static final String INDICATOR_NAME_1 = "1";
+  private static final String INDICATOR_NAME_2 = "2";
+  private static final String SUBFIELD_REGEX = "(?<=\\$).{1}";
+
   private List<Rule> defaultRules;
 
   public List<Rule> create(MappingProfile mappingProfile) {
     if (mappingProfile == null || isEmpty(mappingProfile.getTransformations())) {
       return getDefaultRules();
     }
-    return new ArrayList<>(createByMappingFields(mappingProfile.getTransformations()));
-
+    List<Rule> rules = getDefaultRules();
+    rules.addAll(buildByTransformations(mappingProfile.getTransformations()));
+    return rules;
   }
 
-  private List<Rule> getDefaultRules() {
+  protected List<Rule> getDefaultRules() {
     if (Objects.nonNull(this.defaultRules)) {
       return this.defaultRules;
     }
-    URL url = Resources.getResource("rules/rulesDefault.json");
+    URL url = Resources.getResource(DEFAULT_RULES_PATH);
     String stringRules = null;
     try {
       stringRules = Resources.toString(url, StandardCharsets.UTF_8);
     } catch (IOException e) {
       LOGGER.error("Failed to fetch default rules for export");
       throw new NotFoundException(e);
     }
-    this.defaultRules = Arrays.asList(Json.decodeValue(stringRules, Rule[].class));
+    this.defaultRules = Lists.newArrayList(Json.decodeValue(stringRules, Rule[].class));
     return this.defaultRules;
   }
 
-  // The logic will be replaced in the future
-  private List<Rule> createByMappingFields(List<Transformations> transformations) { //NOSONAR
-    return getDefaultRules();
+  private List<Rule> buildByTransformations(List<Transformations> mappingTransformations) {
+    List<Rule> rules = new ArrayList<>();
+    for (Transformations mappingTransformation : mappingTransformations) {
+      if (TRUE.equals(mappingTransformation.getEnabled()) && isNotBlank(mappingTransformation.getPath())
+        && isNotBlank(mappingTransformation.getTransformation())) {
+        rules.add(buildByTransformation(mappingTransformation));
+      }
+    }
+    return rules;
   }
+
+  private Rule buildByTransformation(Transformations mappingTransformation) {
+    String field = substring(mappingTransformation.getTransformation(), 0, 3);
+    Rule rule = new Rule();
+    rule.setField(field);
+    rule.setDataSources(buildDataSources(mappingTransformation));
+    return rule;
+  }
+
+  private List<DataSource> buildDataSources(Transformations mappingTransformation) {
+    List<DataSource> dataSources = new ArrayList<>();
+    DataSource fromDataSource = new DataSource();
+    fromDataSource.setFrom(mappingTransformation.getPath());
+    dataSources.add(fromDataSource);
+    String transformation = mappingTransformation.getTransformation();
+    Pattern pattern = Pattern.compile(SUBFIELD_REGEX);
+    Matcher matcher = pattern.matcher(transformation);
```

Comment:
Isn't the last character the subfield, why can't you just do a substring there as well? Why do you need a pattern matcher

Reply:
I think in future we will support multiple subfields in one transformation value, therefore it's better to parse it by regex than by position. At the same time inc case if a user adds a lot of spaces it will not fail